dask_kubernetes/operator/deployment/helm/chartpress.yaml

```diff
@@ -21,3 +21,8 @@ charts:
     repo:
       git: dask/helm-chart
       published: https://helm.dask.org
+    images:
+      dask-kubernetes-operator:
+        imageName: ghcr.io/dask/dask-kubernetes-operator
+        dockerfilePath: ../Dockerfile
+        contextPath: ../../../../
```

Comment:
Yeah originally the `Dockerfile` was the only thing in `deployment` and other unrelated stuff has sprouted around it. happy to move it into a `docker` subdirectory if that helps?

We build the docker image on every CI run anyway, so I'm not opposed to this functionality.

Reply:
TL;DR: no change is needed, but I think its a sensible practice to reduce what the Dockerfile interacts with to what is really needs to interact with and also update chartpress.yaml to that reduced scope. This could help with performance in the CI system or to help during local development etc, but its out of scope for this PR.

> We build the docker image on every CI run anyway, so I'm not opposed to this functionality.

`chartpress` behavior is to determine a version/tag for an image based on changes to the docker images context of influence as declared in chartpress.yaml. If this image version/tag is available already, it will pull and re-use it.

Since no dev releases are made, there will not be a already pushed image typically. Especially, if any change at all leads to a change in the image's context, as that leads to a new tag is calculated by chartpress. So, in practice, you will rebuild images all the time no matter what because of the wide image context.

If you reduce the scope of the image context so that the image is rebuilt only if something actually could have influenced it besides unpinned dependencies, it wouldn't rebuild as often etc.

I think it would be a sane choice to isolate the Dockerfile a bit, and make it only use parts of the repo instead of `COPY` the entire root of this project. But the main value of doing so is performance of the CI system in my mind, which won't be great anyhow if dev releases aren't available allowing us to re-use previous image builds.

Reply:
Yeah this is all good advice. The image build here is actually a separate workflow to the test CI, we don't use the image in CI we call the code directly. The main thing I want to test is if the image builds successfully. So we could think about these micro-optimisations in the future, but there are bigger goals right now.